tests/test_agent_tools.py

```
@@ -96,3 +96,95 @@ def test_file_index_respects_max_entries(tmp_path: Path):
+    content = (root / "cfg.py").read_text(encoding="utf-8")
+    assert "DEBUG" not in content
+    assert "PROD" in content
+
+
+def test_edit_file_rejects_path_traversal(tmp_path: Path):
+    root = tmp_path / "repo"
+    root.mkdir()
+    tools = WorkspaceTools(root)
+    try:
+        tools.edit_file("../../etc/passwd", "old", "new")
+        assert False, "Should have raised ValueError"
```

Ruff (0.16.1)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)